File: programmers/k진수에서 소수 개수 구하기.py
def isPrime(num):
    if num == 0 or num == 1:
        return False
    sqr = int(num ** 0.5)
    sqr += 1
    for i in range(2, sqr):
        if num % i == 0:
            return False
    return True


def solution(n, k):
    answer = 0
    changeNum = []
    while n > 0:
        extra = n % k
        changeNum.append(extra)
        n = n // k
    exchangeNum = 0
    digit = 0
    for it in range(len(changeNum)):
        exchangeNum += changeNum[it] * (10 ** digit)
        digit += 1
        if changeNum[it] == 0 or it == len(changeNum) - 1:
            if isPrime(exchangeNum) == True:
                answer += 1
            digit = 0
            exchangeNum = 0

    return answer

File: programmers/k진수에서 소수 개수 구하기.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>

using namespace std;

bool isPrime(long long num){
    if(num == 1 || num == 0)
        return false;
    int sqr = sqrt(num);
    sqr++;
    for(int i = 2; i < sqr; i++){
        if(num % i == 0)
            return false;
    }
    return true;
}

int solution(int n, int k) {
    int answer = 0;
    vector<int> changeNum;
    while(n>0){
        changeNum.push_back(n%k);
        n = n/k;
    }
    long long unitNum = 0;
    int digit = 0;
    for(int i=0; i<changeNum.size(); i++){
        unitNum = unitNum + (changeNum[i] * pow(10, digit));
        digit++;
        if (changeNum[i] == 0 || i == changeNum.size()-1){
            if(isPrime(unitNum))
                answer++;
            unitNum=0;
            digit=0;
        }
    }
    return answer;
}